### PR TITLE
Updating XsmmRunnerUtils w.r.t MatMul changes

### PR DESCRIPTION
**Pull request by nhasabni**
Following changes:

1. Removing _f32 suffix from iree_xsmm_matmul APIs as datatype is an argument to these APIs as well as to tpp.matmul and xsmm.matmul. Datatype can now be passed all the way from MLIR to TPP to XSMM, without the need of datatype specific APIs.

2. Correcting the number of elements and types of params in both dispatch and invoke APIs for MatMul.

*Tests*: These changes work with lowering linalg.matmul into XSMM MatMul via IREE and executing it to produce correct output.

### CODE REVIEW

File: tpp-rt/XsmmRunnerUtils.h [hunk 1]
```diff
@@ -71,14 +71,14 @@ extern "C" MLIR_RUNNERUTILS_EXPORT void _mlir_ciface_xsmm_brgemm_invoke(
 extern "C" MLIR_RUNNERUTILS_EXPORT int
 iree_xsmm_brgemm_dispatch_f32(void *context, void *params, void *reserved);
 extern "C" MLIR_RUNNERUTILS_EXPORT int
-iree_xsmm_matmul_dispatch_f32(void *context, void *params, void *reserved);
+iree_xsmm_matmul_dispatch(void *context, void *params, void *reserved);
```
Comment (rengolin) [inline tpp-rt/XsmmRunnerUtils.h:74]: I imagine this will need to be done for brgemm as well.

**Comment by rengolin** [comment]
> 1. Removing _f32 suffix from iree_xsmm_matmul APIs as datatype is an argument to these APIs as well as to tpp.matmul and xsmm.matmul. Datatype can now be passed all the way from MLIR to TPP to XSMM, without the need of datatype specific APIs.
> 2. Correcting the number of elements and types of params in both dispatch and invoke APIs for MatMul.

That's great! I imagine the API has changed since October, so this is a good change. Can you also test BRGEMM and get those through the same API?

> _Tests_: These changes work with lowering linalg.matmul into XSMM MatMul via IREE and executing it to produce correct output.

Would it be possible to add those tests to our IREE fork? In case we want to merge this, they will definitely require some tests.

File: tpp-rt/XsmmRunnerUtils.h (hunk 1 above)
Comment (nhasabni) [reply tpp-rt/XsmmRunnerUtils.h:74]: Done.

**Comment by rengolin** [comment]
@nhasabni FYI, it's OK to have merge branches on PR, just make sure you squash and merge. We try to not have merge branches on main if possible.

**Pull request merged**